File: Codebase/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Codebase.Models
{
    public class Product
    {
        public int Id { get; set; }
        [MaxLength(100)]
        public required string Name { get; set; }
        public Category? Category { get; set; }
        [MaxLength(300)]
        public string? Description { get; set; }
        public decimal? PriceRubles { get; set; }
        [MaxLength(1000)]
        public string? Notes { get; set; }
        [MaxLength(1000)]
        public string? NotesSpecial { get; set; }
        public string? ImageUrl { get; set; }
        
        public User? UserCreated { get; set; }
        public DateTime? DateTimeCreated { get; set; }
    }
}

File: AngularCatalog.Client/src/models/product.ts
import { Category } from "./category"
import { User } from "./user"

export class Product {
  Id: Number | undefined
  Name!: string
  Category: Category | null | undefined
  Description: string | null | undefined
  PriceRubles: number | null | undefined
  Notes: string | null | undefined
  NotesSpecial: string | null | undefined
  ImageUrl: string | null | undefined
  UserCreated: User | undefined
  DateTimeCreated: Date | undefined
}
